src/components/Canvas/Line.tsx

```diff
@@ -0,0 +1,23 @@
+import { FunctionComponent } from "react";
+import { useContext2d } from "../../hooks/useContext2d";
+import { Vector2 } from "../../models/Vector2";
+
+type LineProps = {
+    start: Vector2;
+    end: Vector2;
+    color?: string;
+};
+
+export const Line: FunctionComponent<LineProps> = ({ start, end, color }) => {
+    const context = useContext2d();
+
+    if (context !== null) {
+        context.beginPath();
+        context.moveTo(start[0], start[1]);
+        context.lineTo(end[0], end[1]);
+        context.strokeStyle = color || "#000000";
+        context.stroke();
+    }
+
+    return null;
```

Comment:
Is this `null` return intended?

Reply:
I mainly just don't want to actually render any DOM element since all of these are represented via canvas instructions. Not sure if theres a more appropriate way to do that or if just returning nothing is more conventional

Reply:
Maybe that's a clue that a functional component isn't what's really wanted here, but I'm not sure what the right solution is. I'm fine with this but it definitely feels a little odd.

Reply:
You could roll the whole thing into the canvas component but I figure converting into a tree-like structure might be more helpful think about. Ie: you draw things to the canvas by putting child elements into the canvas component. Technically they should probably be siblings of the canvas since the child of the real canvas component is shown by the browser if your browser doesn't support the canvas element... for some reason. In this scenario nothing would get shown because all of these elements should return null (or something like it).